crates/common/fork_choice/src/fork_choice/fork_choice_handlers.rs

```diff
@@ -0,0 +1,217 @@
+use alloy_primitives::B256;
+use anyhow::{bail, ensure};
+use ream_consensus::{
+    checkpoint::Checkpoint,
+    deneb::beacon_block::SignedBeaconBlock,
+    execution_engine::{blob_versioned_hashes::blob_versioned_hashes, engine_trait::ExecutionApi},
+    fork_choice::{
+        constants::{INTERVALS_PER_SLOT, SECONDS_PER_SLOT},
+        store::Store,
+    },
+    kzg_commitment::KZGCommitment,
+    misc::{compute_epoch_at_slot, compute_start_slot_at_epoch},
+};
+use ream_polynomial_commitments::polynomial_commitments::polynomial_commitments_handlers::verify_blob_kzg_proof_batch;
+use tree_hash::TreeHash;
+
+pub async fn is_data_available(
+    beacon_block_root: B256,
+    store: Store,
+    blob_kzg_commitments: Vec<KZGCommitment>,
+    execution_engine: &impl ExecutionApi,
+) -> anyhow::Result<bool> {
+    // `retrieve_blobs_and_proofs` is implementation and context dependent
+    // It returns all the blobs for the given block root, and raises an exception if not available
+    // Note: the p2p network does not guarantee sidecar retrieval outside of
+    // `MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS`
+
+    let Some(beacon_block) = store.blocks.get(&beacon_block_root) else {
+        bail!("could not get beack block");
+    };
+    let blob_versioned_hashes = blob_versioned_hashes(&beacon_block.body.execution_payload)?;
+    let blobs_and_proofs = execution_engine
+        .engine_get_blobs_v1(blob_versioned_hashes)
+        .await?;
+
+    let mut blobs = vec![];
+    let mut proofs = vec![];
+
+    for block_and_proof in blobs_and_proofs {
+        let block_and_proof =
+            block_and_proof.ok_or_else(|| anyhow::anyhow!("Invalid proposer index"))?;
+        blobs.push(block_and_proof.blob);
+        proofs.push(block_and_proof.proof);
+    }
+
+    verify_blob_kzg_proof_batch(&blobs, blob_kzg_commitments, &proofs)?;
+    Ok(true)
+}
+
+pub fn get_ancestor(store: Store, root: B256, slot: u64) -> B256 {
+    let binding = store.blocks.clone();
+    if let Some(block) = binding.get(&root) {
```

Comment:
```suggestion
pub fn get_ancestor(store: &Store, root: B256, slot: u64) -> B256 {
    if let Some(block) = store.blocks.get(&root) {
```

we need to avoid the clone here